extensions/samples/electron-webpack/src/main.ts

```diff
@@ -1,92 +1,89 @@
+import AuthProvider from "./AuthProvider";
+import { authConfig } from "./authConfig";
 
-const persistenceConfiguration = {
-  cachePath,
-  dataProtectionScope: DataProtectionScope.CurrentUser,
-  serviceName: "test-msal-electron-service",
-  accountName: "test-msal-electron-account",
-  usePlaintextFileOnLinux: false,
-}
+declare const MAIN_WINDOW_WEBPACK_ENTRY: string;
+declare const MAIN_WINDOW_PRELOAD_WEBPACK_ENTRY: string;
 
-function createWindow() {
-  // Create the browser window.
-  const mainWindow = new BrowserWindow({
-    height: 600,
-    webPreferences: {
-      nodeIntegration: true,
-      contextIsolation: false,
-    },
-    width: 800,
-  });
+export default class Main {
+    static application: Electron.App;
+    static mainWindow: Electron.BrowserWindow;
+    static authProvider: AuthProvider;
+    static authConfig: any;
 
-  PersistenceCreator
-  .createPersistence(persistenceConfiguration)
-  .then(async (persistence: IPersistence) => {
-    // Initialize the electron authenticator
-    const config: MsalElectronConfig = {
-      clientId: '<CLIENT_ID>',
-      mainWindow: mainWindow, //BrowserWindow instance that requires auth
-      scopes: [
-        'user.read',
-        'people.read',
-        'user.readbasic.all',
-        'contacts.read',
-        'presence.read.all',
-        'presence.read',
-        'user.read.all',
-        'calendars.read',
-        'Sites.Read.All',
-        'Sites.ReadWrite.All',
-      ],
-      cachePlugin: new PersistenceCachePlugin(persistence),
-    };
+    static main(): void {
+        Main.application = app;
+        Main.application.on("window-all-closed", Main.onWindowAllClosed);
+        Main.application.on("ready", Main.onReady);
+        Main.authConfig = authConfig;
+    }
 
-    ElectronAuthenticator.initialize(config);
+    private static onWindowAllClosed(): void {
+        // Windows and Linux will quit the application when all windows are closed
+        if (process.platform !== "darwin") {
+            Main.application.quit();
+        }
+    }
 
-    // and load the index.html of the app.
-    mainWindow.loadFile(path.join(__dirname, "../index.html"));
+    private static async onReady(): Promise<void> {
+        try {
+            Main.createMainWindow();
+            Main.mainWindow.loadURL(MAIN_WINDOW_WEBPACK_ENTRY);
+            Main.mainWindow.on("closed", Main.onClose);
+            Main.authProvider = await AuthProvider.build(Main.authConfig);
+            Main.registerSubscriptions();
+        } catch (error) {
+            console.log(error);
+        }
+    }
 
-    // Open the DevTools.
-    mainWindow.webContents.openDevTools();
-  });
-}
+    private static createMainWindow(): void {
+        this.mainWindow = new BrowserWindow({
+            width: 800,
+            height: 600,
+            webPreferences: {
+                preload: MAIN_WINDOW_PRELOAD_WEBPACK_ENTRY,
+                nodeIntegration: false,
```

Comment:
> nodeIntegration: false

Since this is by default false in the current electron version, I think we can remove it.

Reply:
Done!